app/create.py

```diff
@@ -37,6 +37,254 @@ def days_to_date(days):
+    else:
+        if not given_name:
+            raise CreationError(_('Name is required.  Please go back and try again.'))
+
+    # If user is inputting his/her own information, set some params automatically
+    if own_info == 'yes':
+        author_name = given_name
+        status = 'is_note_author'
+        author_made_contact = 'yes'
+        if users_own_email:
+            author_email = users_own_email
+        if users_own_phone:
+            author_phone = users_own_phone
+
+    if (author_email and not validate_email(author_email)):
+        raise CreationError(_('The email address you entered appears to be invalid.'))
+
+    else:
+        if not author_name:
+            if clone:
+                raise CreationError(_('The Original author\'s name is required.  Please go back and try again.'))
+            else:
+                raise CreationError(_('Your name is required in the "Source" section.  Please go back and try again.'))
+
+    if add_note:
+        if not text:
+            raise CreationError(_('Message is required. Please go back and try again.'))
+        if status == 'is_note_author' and not author_made_contact:
+            raise CreationError(_('Please check that you have been in contact with the person after the earthquake, or change the "Status of this person" field.'))
+        if status == 'believed_dead' and not config.allow_believed_dead_via_ui:
+            raise CreationError(_('Not authorized to post notes with the status "I have received information that this person is dead".'))
+
+    if source_date:
+        try:
+            source_date = validate_date(source_date)
+        except ValueError:
+            raise CreationError(_('Original posting date is not in YYYY-MM-DD format, or is a nonexistent date.  Please go back and try again.'))
+        if source_date > now:
+            raise CreationError(_('Date cannot be in the future.  Please go back and try again.'))
+
+    expiry_date = days_to_date(expiry_option or config.default_expiry_days)
+
+    profile_urls = filter(lambda url: url, profile_urls)
+    url_validator = URLValidator(schemes=['http', 'https'])
+    for profile_url in profile_urls:
+        try:
+            url_validator(profile_url)
+        except ValidationError:
+            raise CreationError(_('Please only enter valid profile URLs.'))
+
+    # If nothing was uploaded, just use the photo_url that was provided.
+    try:
+        # If a photo was uploaded, create a Photo entry and get the URL
+        # where we serve it.
+        if photo is not None:
+            photo, photo_url = create_photo(photo, repo, url_builder)
+        if note_photo is not None:
+            note_photo, note_photo_url = create_photo(
+                note_photo, repo, url_builder)
+    except PhotoError, e:
+        raise CreationError(_('There seems to be a problem with your uploaded photo.'))
```

Comment:
just checking, is it worth specifying which photo/note_photo failed? (in case they can upload both?)

Reply:
I don't think we ever give users a form with both options, so they should only be providing one or the other. (Related: just filed issue #696 because I'm not sure this needs to handle a `note_photo` field at all.)